src/passes/OptimizeInstructions.cpp

```diff
@@ -301,6 +294,19 @@ struct OptimizeInstructions
           return sub;
         }
       }
+      {
+        // eqz((signed)x % C_pot)  =>  eqz(x & (C_pot - 1))
+        Const* c;
+        Binary* inner;
+        if (matches(curr,
+                    unary(Abstract::EqZ,
+                          binary(&inner, Abstract::RemS, any(), ival(&c)))) &&
+            IsPowerOf2((uint64_t)c->value.getInteger())) {
```

Comment:
why is the cast to `unit64_t` needed?

Reply:
`IsPowerOf2` allow only unsigned types but `getInteger()` return `int64_t`

Reply:
Btw for now `x % -4 == 0` rejected and don't optimize which is ok but it seems it could be handle as `(x & 3) == 0` as well.

Reply:
Wouldn't it work without the cast, then? The function is
```cpp
template<typename T> bool isPowerOf2(T v) {
  return v != 0 && (v & (v - 1)) == 0;
}
```
which just cares about bits anyhow. There should be no compiler warning or error here? Maybe I'm not sure what you mean by "allow only".

Reply:
I'm not sure signed type does make sense for `isPowerOf2` due to negative values always return `false`

Reply:
Logically, I agree, it's odd to ask about a negative number. But we just have bits here, really. A Literal just stores the bits, without a sign.

Reply:
I'll create PR which will handle also negative dividers: `x % -C_pot` -> `x & (abs(C_pot) - 1)`